src/traces/pie/plot.js

```diff
@@ -95,7 +95,19 @@ module.exports = function plot(gd, cdpie) {
                     // in case fullLayout or fullData has changed without a replot
                     var fullLayout2 = gd._fullLayout;
                     var trace2 = gd._fullData[trace.index];
-                    var hoverinfo = Fx.castHoverinfo(trace2, fullLayout2, pt.i);
+
+                    var hoverinfo = trace2.hoverinfo;
+                    if(Array.isArray(hoverinfo)) {
+                        // super hacky: we need to pull out the *first* hoverinfo from
+                        // pt.pts, then put it back into an array in a dummy trace
+                        // and call castHoverinfo on that.
+                        // TODO: do we want to have Fx.castHoverinfo somehow handle this?
+                        // it already takes an array for index, for 2D, so this seems tricky.
+                        hoverinfo = Fx.castHoverinfo({
+                            hoverinfo: [helpers.castOption(hoverinfo, pt.pts)],
+                            _module: trace._module
+                        }, fullLayout2, 0);
```

Comment:
@etpinard thoughts? I didn't see a reasonable way to build this into `Fx.castHoverInfo` since an array index already has a meaning there...

Reply:
Maybe something like:

```js
Fx.castHoverinfo = function(trace, fullLayout, pt) {
   if(pt.pointNumber) {
     // as currently
   } else if(pt.pointNumbers) {
     // new pie logic
   }
}

// instead of
Fx.castHoverinfo = function(trace, fullLayout, ptNumber) {
  // ...
}
```

but your hack (with that comment) is fine :+1:

Reply:
I think I should leave it as is for now, but if we unify this behavior with histograms (see #1984) then it may be worth making this change.